contrib/src/main/scala/akka/stream/contrib/PartitionWith.scala

```diff
@@ -120,6 +133,18 @@ final class PartitionWith[In, Out0, Out1] private (p: In => Either[Out0, Out1])
         }
       }
       else if (!hasBeenPulled(in)) pull(in)
+
+      override def onDownstreamFinish(): Unit =
+        downstreamFinished()
     })
+
+    private def downstreamFinished(): Unit = {
+      activeDownstreamCount -= 1
+      if (eagerCancel) {
+        completeStage()
+      } else if (activeDownstreamCount == 0) {
+        cancel(in)
```

Comment:
Shouldn't it be `completeStage()` here as well?

Reply:
Does it make any difference? By this time, both downstreams have finished, so it remains to only signal cancellation to the upstream.

Reply:
Nah, maybe not.